source/MaterialXContrib/MaterialXNode/MaterialXData.cpp

```diff
@@ -53,51 +106,26 @@ const std::string& MaterialXData::getFragmentName() const
     return _fragmentName;
 }
 
-const std::string& MaterialXData::getFragmentWrapper() const
+const std::string& MaterialXData::getFragmentSource() const
 {
-    return _fragmentWrapper;
+    return _fragmentSource;
 }
 
 const mx::StringMap& MaterialXData::getPathInputMap() const
 {
     return _pathInputMap;
 }
 
-void MaterialXData::createDocument(const std::string& materialXDocumentPath)
-{
-    // Create document
-    _document = mx::createDocument();
-
-    // Load libraries
-    static const mx::StringVec libraries = { "stdlib", "pbrlib", "bxdf", "stdlib/genglsl", "pbrlib/genglsl" };
-    MaterialXMaya::loadLibraries(libraries, _librarySearchPath, _document);
-
-    // Read document contents from disk
-    mx::XmlReadOptions readOptions;
-    readOptions.skipDuplicateElements = true;
-    mx::readFromXmlFile(_document, materialXDocumentPath, mx::EMPTY_STRING, &readOptions);
-}
-
 bool MaterialXData::elementIsAShader() const
 {
-    return (_element ? _element->isA<mx::ShaderRef>() : false);
-}
-
-void MaterialXData::clearXml()
```

Comment:
Removing the clear method because as we discussed today we're going to rely on MaterialXData destruction when reloading the document or switching the element.